mapper/mapper_test.go

```diff
@@ -40,4 +40,33 @@ func TestUnitMapper(t *testing.T) {
 		So(len(mapped), ShouldEqual, 2)
 	})
 
+	Convey("that datasets are ordered alphabetically by Title", t, func() {
+		ds := dataset.List{
+			Items: []dataset.Dataset{},
+		}
+		ds.Items = append(ds.Items, dataset.Dataset{
+			ID: "test-id-3",
+			Next: &dataset.DatasetDetails{
+				Title: "3rd Title",
+			},
+		}, dataset.Dataset{
+			ID: "test-id-1",
+			Next: &dataset.DatasetDetails{
+				Title: "1st Title",
+			},
+		}, dataset.Dataset{
+			ID: "test-id-2",
+			Next: &dataset.DatasetDetails{
+				Title: "2nd Title",
+			},
+		})
+
+		mapped := AllDatasets(ds)
+
+		So(mapped[0].Title, ShouldEqual, "1st Title")
+		So(mapped[1].Title, ShouldEqual, "2nd Title")
+		So(mapped[2].Title, ShouldEqual, "3rd Title")
```

Comment:
Looks good but for full marks I think you should add a case where the title is empty to cover off the `if` check in your code.

Reply:
Good shout. Updated the test cases and the logic a little off the back of that